### ISSUE TITLE
Issue with options and shield with a name

### ISSUE THREAD

**Issue by Katalijst** [SkriptLang/Skript#2534]
### Description
When using an option referring to a shield with a name, it doesn't recognise the shield as an item if you write "shield" and not "shields".

### Steps to Reproduce
```
options:
	admin_shield: shield named "&fAdmin Shield"

command /AdminShield:
	trigger:
		give player {@admin_shield}
```

use this code with skript 2.4-beta8, and in game do /adminshield, not tested on other version

### Expected Behavior
I should get a shield with the name "Admin Shield" using "shield" or "shields"

### Errors / Screenshots
No error in the console

### Server Information
* **Server version/platform:**      Paper version git-Paper-209 (MC: 1.14.4)
* **Skript version:**                Skript 2.4-beta8

**Comment by Katalijst** [comment]
> ### Description
> When using an option referring to a shield with a name, it doesn't recognise the shield as an item.
> 
> ### Steps to Reproduce
> ```
> options:
> 	admin_shield: shield named "&fAdmin Shield"
> 
> command /AdminShield:
> 	trigger:
> 		give player {@admin_shield}
> ```
> 
> use this code with skript 2.4-beta8, and in game do /adminshield, not tested on other version
> 
> ### Expected Behavior
> I should get a shield with the name "Admin Shield"
> 
> ### Errors / Screenshots
> No error in the console
> 
> ### Server Information
> * **Server version/platform:**      Paper version git-Paper-209 (MC: 1.14.4)
> * **Skript version:**                Skript 2.4-beta8

**Comment by ShaneBeee** [comment]
This has nothing to do with options, it has to do with item types with data.
ex:
`give player diamond sword named "bob"` will not work
`give player a diamond sword named "bob"` will work

You need to add `a` before the item

**Comment by Katalijst** [comment]
But then why does it works with "shields" and not "shield" ?

**Comment by Wealthyturtle** [comment]
That's grammar... look at the 2 following sentences:
Bob gave **a apple** to Steve
Bob gave **apples** to Steve

**Comment by Blueyescat** [comment]
If there is no error it should definitely work (also the code is correct). I think Shane states that there is a bug with `shield` but `a shield` works (weird)

**Comment by Wealthyturtle** [comment]
Hmm... What I was able to find is:
```
command /test:
	trigger:
		give player shield
```
Works

```
command /test:
	trigger:
		give player shield named "test"
```
Doesn't Work

Both parse successfully, but the one with the ``named "test"`` doesn't

**Comment by Katalijst** [comment]
Wealthyturtle I know that's grammar but both should work the same according to the alias files

**Comment by bensku** [comment]
In second case, item is from expression. Aliases are not expressions; that could explain the difference.

**Comment by ShaneBeee** [comment]
Okay... after much MUCH testing I figured out what the issue is. 

When the named expression is initializing, its thinking that `shield` is the player's shield.
I ran two tests, one with `shield` and one with `a shield`, outcome: (I just added some debug code into the plugin to test the initialization)
```
[00:21:16 INFO]: This is initializing - name: "shield test" exp: the tool of event-living entity
[00:21:16 INFO]: This is initializing - name: "a shield test" exp: shield
```
as you can see `shield` returns the expression `the tool of event-living entity`
and `a shield` returns the expression `shield`

I'm not entirely sure how to fix this, but at least I found where the bug is coming from!

**Comment by bluelhf** [comment]
Aliases could take priority over expressions if the expression directly matches an alias?

shield -> alias, because shield exists as one
player's shield -> expression, because there is no `player's shield` alias

**Comment by TPGamesNL** [comment]
Fixed in #4002
